packages/issuer/src/blockchain-facade/Certificate.ts

```diff
@@ -0,0 +1,257 @@
+
+    issuer: string;
+    
+    deviceId: string;
+    generationStartTime: number;
+    generationEndTime: number;
+
+    energy: number;
+    creationTime: number;
+
+    sync(): Promise<ICertificate>;
+
+    getCertificateOwner(): Promise<string>;
+    isClaimed(): Promise<boolean>;
+    claim(amount: number): Promise<TransactionReceipt>;
+
+    transfer(to: string, amount: number): Promise<TransactionReceipt>;
+
+    getAllCertificateEvents(): Promise<EventLog[]>;
+}
+
+const getAccountFromConfiguration = (configuration: Configuration.Entity) => ({
+    from: configuration.blockchainProperties.activeUser.address,
+    privateKey: configuration.blockchainProperties.activeUser.privateKey
+});
+
+export const getAllCertificateEvents = async (
+    certId: number,
+    configuration: Configuration.Entity
+): Promise<EventLog[]> => {
+    const registry: Registry = configuration.blockchainProperties.certificateLogicInstance;
+
+    const allEvents = await registry.getAllEvents(
+        {
+            topics: [
+                null,
+                configuration.blockchainProperties.web3.utils.padLeft(
+                    configuration.blockchainProperties.web3.utils.fromDecimal(certId),
+                    64,
+                    '0'
+                )
+            ],
+            toBlock: undefined
+        }
+    );
+
+    const returnEvents = [];
+
+    for (const fullEvent of allEvents) {
+        // we have to remove some false positives due to ERC721 interface
+        if (fullEvent.event === 'Transfer') {
+            if (fullEvent.returnValues.tokenId === `${certId}`) {
+                returnEvents.push(fullEvent);
+            }
+        } else {
+            returnEvents.push(fullEvent);
+        }
+    }
+
+    // we also have to search
+    if (certId !== 0) {
+        const transferEvents = await registry.getAllTransferSingleEvents(
+            {
+                topics: [
+                    '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef',
+                    null,
+                    null,
+                    configuration.blockchainProperties.web3.utils.padLeft(
+                        configuration.blockchainProperties.web3.utils.fromDecimal(certId),
+                        64,
+                        '0'
+                    )
+                ],
+                toBlock: undefined
+            }
+        );
+
+        for (const transferEvent of transferEvents) {
+            returnEvents.push(transferEvent);
+        }
+    }
+
+    return returnEvents;
+};
+
+export class Entity extends BlockchainDataModelEntity.Entity implements ICertificate {
+    public deviceId: string;
+
+    public generationStartTime: number;
+    public generationEndTime: number;
+    public issuer: string;
+    public energy: number;
+    public creationTime: number;
+
+    public initialized: boolean;
+
+    private data: number[];
+
+    constructor(id: string, configuration: Configuration.Entity) {
```

Comment:
Constructor could be removed if `initialized` declaration would declare its value as well.

Aka

`public initialized = false`

Reply:
Not really, the constructor also calls `super(id, configuration);`